lang/cs/Org.Apache.REEF.Common/TeleMetry/Counters.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    public class Counters : ICounters
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(Counters));
+
+        /// <summary>
+        /// It contains name and count pairs
+        /// </summary>
+        private readonly IDictionary<string, int> _counters = new Dictionary<string, int>();
+
+        /// <summary>
+        /// The lock for counters
+        /// </summary>
+        private readonly object _counterLock = new object();
+
+        [Inject]
+        private Counters()
+        {
+        }
+
+        /// <summary>
+        /// register an new counter with a specified name. 
+        /// If name does not exist, the counter will be added and true will be returned
+        /// Otherwise the counter will be not added and false will be returned. 
+        /// </summary>
+        /// <param name="name">Counter name</param>
+        /// <returns></returns>
+        public bool TryRegisterCounter(string name)
+        {
+            lock (_counterLock)
+            {
+                if (_counters.ContainsKey(name))
+                {
+                    Logger.Log(Level.Warning, "The counter [{0}]  already exists.", name);
+                    return false;
```

Comment:
User can choose to ignore the return value. But if it is duplicated with one in system, the user would have chance to use another name when false is returned.